File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = defaultdict(lambda: -1)
        
        for idx, n in enumerate(nums):
            comp = target - n
            if seen[comp] >= 0:
                return [idx, seen[comp]]
            seen[n] = idx

File: turingplanet/hashtable/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

#include "lib/headers.h"

// @lc code=start
class Solution
{
  public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++)
        {
            int remain = target - nums[i];
            if (seen.count(remain))
                return {seen[remain], i};
            seen[nums[i]] = i;
        }
        return {};
    }
};
// @lc code=end
